src/Functions/FunctionsCoding.h

```diff
@@ -1506,6 +1507,95 @@ class FunctionBitmaskToArray : public IFunction
     }
 };
 
+class FunctionBitPositionsToArray : public IFunction
+{
+public:
+    static constexpr auto name = "bitPositionsToArray";
+    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionBitPositionsToArray>(); }
+
+    String getName() const override
+    {
+        return name;
+    }
+
+    size_t getNumberOfArguments() const override { return 1; }
+    bool isInjective(const ColumnsWithTypeAndName &) const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (!isNativeInteger(arguments[0]))
+            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "Illegal type {} of argument of function {}",
+                getName(),
+                arguments[0]->getName());
+
+        return std::make_shared<DataTypeArray>(std::make_shared<DataTypeUInt64>());
+    }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    template <typename T>
+    ColumnPtr executeType(const IColumn * column) const
+    {
+        const ColumnVector<T> * col_from = checkAndGetColumn<ColumnVector<T>>(column);
+        if (!col_from)
+            return nullptr;
+
+        auto result_array_values = ColumnVector<UInt64>::create();
+        auto result_array_offsets = ColumnArray::ColumnOffsets::create();
+
+        auto & result_array_values_data = result_array_values->getData();
+        auto & result_array_offsets_data = result_array_offsets->getData();
+
+        auto & vec_from = col_from->getData();
+        size_t size = vec_from.size();
+        result_array_offsets_data.resize(size);
```

Comment:
I am not sure, but isn't it more effective to do `reserve` + `push_back`?

Reply:
resize + operator[] is more efficient.

Reply:
I always though that `reserve` + `push_back` is more efficient...
I made a small test.
`reserve` + `push_back`:
```c++
int main()
{
    std::vector<int> v;
    v.reserve(100000000);
    for (int i = 0; i != 100000000; ++i)
        v.push_back(i);
    return 0;
}
```
If compile using clang++-11 and option -O2 I get:
```
real	0m0.278s
user	0m0.073s
sys	0m0.205s
```

`resize` + `operator[]`:
```c++
int main()
{
    std::vector<int> v;
    v.resize(100000000);
    for (int i = 0; i != 100000000; ++i)
        v[i] = i;
    return 0;
}
```
Time:
```
real	0m0.312s
user	0m0.108s
sys	0m0.203s
```
Can you explain why `resize` + `operator[]` is more efficient?

Reply:
Better to check with -O3 and look at the generated code. Also replace index to size_t.
It should be better due to absense of overflow check (it is predictable branch, so I don't expect much difference in runtime).

Reply:
@alexey-milovidov, @avogar need to test with PODArray because vector resize initialize elements.
```c++
int main(int argc, char **argv)
{
    (void)(argc);
    (void)(argv);

    size_t insert_size = std::stoull(argv[1]);
    size_t result = 0;

    DB::PaddedPODArray<size_t> values;
    values.resize(insert_size);

    for (size_t i = 0; i < insert_size; ++i)
    {
        values[i] = i;
        ++result;
    }

    return result;
}
```
time ./pod_array_example 100000000
```
real	0m0.214s
user	0m0.020s
sys	0m0.194s
```
```c++
int main(int argc, char **argv)
{
    (void)(argc);
    (void)(argv);

    size_t insert_size = std::stoull(argv[1]);
    size_t result = 0;

    DB::PaddedPODArray<size_t> values;
    values.reserve(insert_size);

    for (size_t i = 0; i < insert_size; ++i)
    {
        values.push_back(insert_size);
        ++result;
    }

    return result;
}
```

time ./pod_array_example 100000000
```
real	0m0.251s
user	0m0.076s
sys	0m0.175s
```

Reply:
Yes, I was thinking about PODArray :)